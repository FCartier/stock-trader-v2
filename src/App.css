/* Global Variables */
@import url("https://fonts.googleapis.com/css?family=Lato:300,400");
/* Colors */
/* Essentials */
.logo {
  width: 10em;
  height: auto;
}

html, body {
  height: 100%;
  width: 100%;
  margin: 0;
}

body {
  min-height: 100%;
  margin: 0;
  font-family: "Lato", sans-serif;
  color: white;
}

.gradient {
  background-image: linear-gradient(to bottom right, #1E355E, #16233B);
  min-height: 100vh;
}

.category-header {
  color: #72a2e9;
  border-bottom: 2px solid #72a2e9;
}

.category-header > h3 {
  margin-bottom: 0px;
  padding-bottom: 0px;
}

.market-foot {
  display: flex;
  font-size: 10px;
  list-style-type: none;
  align-items: stretch;
  justify-content: space-between;
  max-width: 100%;
  margin-bottom: 1rem;
  background-image: linear-gradient(to bottom right, #01255c, #002050);
  overflow: hidden;
}

footer {
  flex-shrink: 0;
}

/* Grid + Responsive */
.flexbox-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-left: 2em;
  margin-right: 2em;
}

.main-container {
  display: flex;
}

.sidebar {
  order: 1;
  flex: 1;
  flex-basis: auto;
  min-height: 666px;
  margin-right: 3em;
  margin-left: 3em;
}

.main {
  margin-top: 2em;
  min-width: 70%;
}

.sidebar-column {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: inherit;
}

.box {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: inherit;
}

.Grid {
  display: flex;
  flex-flow: row;
  flex-wrap: wrap;
}

.Grid-cell {
  flex: 1;
}

.Grid--gutters {
  margin-left: -1em;
}
.Grid--gutters .Grid-cell {
  padding-left: 1em;
}
.Grid--gutters .Grid--nested .Grid-cell:first-of-type .Demo {
  margin-right: 1em;
}

.Grid--1of2 > .Grid-cell,
.Grid--1of4 > .Grid-cell:first-of-type,
.Grid--1of3 > .Grid-cell:first-of-type {
  flex: 0 0 100%;
}

@media (min-width: 30em) {
  .Grid--1of2 > .Grid-cell {
    flex: 0 0 50%;
  }

  .Grid--1of6 > .Grid-cell:first-of-type {
    flex: 0 0 30%;
  }

  .Grid--1of4 > .Grid-cell:first-of-type {
    flex: 0 0 50%;
  }

  .Grid--1of3 > .Grid-cell:first-of-type {
    flex: 0 0 100%;
  }
}
@media (min-width: 48em) {
  .Grid--1of2 > .Grid-cell {
    flex: 0 0 50%;
  }

  .Grid--1of6 > .Grid-cell:first-of-type {
    flex: 0 0 16.6666%;
  }

  .Grid--1of4 > .Grid-cell:first-of-type {
    flex: 0 0 25%;
  }

  .Grid--1of3 > .Grid-cell:first-of-type {
    flex: 0 0 30%;
  }

  .Grid--gutters.Grid--nested .Grid-cell:first-of-type .Demo {
    margin-right: 0;
  }
}
@media (min-width: 200px) and (max-width: 1060px) {
  .flexbox-container {
    flex-wrap: wrap;
  }

  .keystats-row {
    margin-bottom: 30px;
  }
}
@media all and (max-width: 1040px) {
  .keystats-row {
    display: block;
  }
}
@media all and (min-width: 1040px) {
  .keystats-row {
    display: flex;
  }
}
@media all and (max-width: 760px) {
  .news {
    margin-top: 4rem;
  }

  .news,
.overview,
.top-peers {
    margin-left: 1rem;
  }

  .overview-content {
    padding: 1rem;
  }

  .keystats-row {
    padding-right: 1rem;
  }
}
/* App Components */
.keystats-column {
  flex: 50%;
}

.keystats-list {
  list-style-type: none;
  padding-left: 1em;
  padding-right: 1em;
}

.keystats-list > li {
  border-bottom: 2px solid grey;
}

.keystats-list > li > p {
  display: inline-block;
  width: 49%;
}

.keystats-list > li > h5 {
  display: inline-block;
  text-align: right;
  padding: 0px;
  margin: 0px;
  width: 50%;
}

.keystats {
  margin-left: 1em;
}

.news-list {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.article-header {
  margin: 0px;
  padding: 0px;
}

.article-footer {
  padding-bottom: 20px;
  padding-top: 4px;
  margin: 0px;
  font-size: 9px;
}

.article-header > a {
  text-decoration: none;
  color: white;
}

.news-list > li {
  padding: 1rem;
}

.price {
  font-size: 2em;
  display: inline-block;
}

.text-orange {
  color: #df5455;
  display: inline-block;
}

.text-green {
  color: #6fb792;
  display: inline-block;
}

sup {
  font-size: 0.5em;
}

.searchbar {
  width: 80%;
  height: 4em;
}

.searchbar-container {
  align-items: center;
  font-weight: 300;
  justify-items: space-between;
}

.Select-control {
  background-color: transparent;
  color: #fff;
  border: 0;
  width: 80%;
  height: 100%;
}

.Select.is-focused > .Select-control, .Select.is-focused:not(.is-open) > .Select-control {
  background: transparent;
  color: #fff;
  border-color: transparent;
}

.Select.has-value.Select--single > .Select-control .Select-value .Select-value-label, .Select.has-value.is-pseudo-focused.Select--single > .Select-control .Select-value .Select-value-label {
  color: #fff;
}

.Select-menu-outer {
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
  background-color: #16233B;
  border: 0;
  box-shadow: 0 1px 0 rgba(0, 0, 0, 0.06);
  box-sizing: border-box;
  margin-top: -1px;
  max-height: 200px;
  width: 70%;
  position: absolute;
  left: 0;
  top: 100%;
  width: 100%;
  z-index: 1;
}

.Select {
  color: #fff;
}

.full-height, .Select {
  height: 100%;
}

.full-height {
  justify-items: center;
}

.search-icon {
  width: 1.5em;
  height: auto;
}

.Select-placeholder, .Select--single > .Select-control .Select-value {
  font-size: 2em;
}

.icon-container {
  display: inline-block;
}

.searchbox-container {
  display: inline-block;
  min-width: 80%;
}

.top-peers-list {
  display: flex;
  align-items: stretch;
  justify-content: space-between;
  width: 100%;
  margin: 0;
  padding: 0;
}

.top-peers-element {
  display: block;
  flex: 0 1 auto;
  list-style-type: none;
}

.chartContainer {
  height: 35em;
}

.chart-filter {
  text-align: right;
  margin-right: 90px;
}

.chart-filter > button {
  background-color: Transparent;
  background-repeat: no-repeat;
  border: none;
  overflow: hidden;
  color: grey;
}

.referenceLine > label {
  position: "right";
  fill: blue;
  font-size: 14;
  text-align: "right";
}

.keystats-column {
  flex: 50%;
}

.keystats-list {
  list-style-type: none;
  padding-left: 1em;
  padding-right: 1em;
}

.keystats-list > li {
  border-bottom: 2px solid grey;
}

.keystats-list > li > p {
  display: inline-block;
  width: 49%;
}

.keystats-list > li > h5 {
  display: inline-block;
  text-align: right;
  padding: 0px;
  margin: 0px;
  width: 50%;
}

.keystats {
  margin-left: 1em;
}

/*# sourceMappingURL=App.css.map */
