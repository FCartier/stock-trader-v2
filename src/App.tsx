import React from "react";
import "babel-polyfill";
import News from "./components/news";
import Overview from "./components/overview";
import KeyStats from "./components/keystats";
import Chart from "./components/chart";
import TopPeers from "./components/peers";
import MarketFooter from "./components/footer";
import Price from "./components/price";
import Search from "./components/search/Search";
import { Gradient } from "./globalStyles";
import { Logo } from "./globalStyles";
import * as logo from "./images/logo.png";
import { ThemeProvider } from 'emotion-theming'
import { darkTheme, lightTheme } from './themes'
import { ThemeChangeButton } from './globalStyles'
import "./grid.css";



class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isDark: true,
      theme: darkTheme
    };
  }

  handleClick() {
    const isDark = !this.state.isDark;
    this.setState({
      isDark,
      theme: isDark ? darkTheme : lightTheme
    });
  }

  render() {
    return (
      <ThemeProvider theme={this.state.theme}>
        <Gradient>
          <div className="flexbox-container">
            <div>
              <Logo src={logo} alt="Adaptive Logo" />
              <ThemeChangeButton onClick={() => this.handleClick()}>Change Theme</ThemeChangeButton>
            </div>
          </div>

          <div className="flexbox-container searchbar-container">
            <div className="searchbar">
              <Search />
            </div>
            <div className="price-information">
              <Price />
            </div>
          </div>

          <hr />

          <div className="Grid Grid--gutters Grid--1of3">
            <div className="Grid-cell main">
              <div>
                <Chart />
              </div>
            </div>

            <div className="Grid-cell">
              <div>
                <News />
              </div>
            </div>
          </div>

          <div className="Grid Grid--gutters Grid--1of3">
            <div className="Grid-cell main">
              <KeyStats />
            </div>

            <div className="Grid-cell">
              <div>
                <Overview />
                <TopPeers />
              </div>
            </div>
          </div>
          <div>
            <footer className="footer">
              <MarketFooter />
            </footer>
          </div>
        </Gradient>
      </ThemeProvider>
    );
  }
}

export default App;
