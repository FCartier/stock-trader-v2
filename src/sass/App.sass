/* Global Variables */
@import "./components/global"
  
/* Essentials */
@import "./components/header"
@import "./components/base"


/* Grid + Responsive */
@import "./components/grid"
@import "./components/media-queries"


/* App Components */

@import "./components/search"
