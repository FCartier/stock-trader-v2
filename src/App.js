import React, { Component } from 'react'
import 'babel-polyfill';
import './App.css'
import News from './components/news'
import Overview from './components/overview'
import KeyStats from './components/keystats/KeyStats.tsx'
import ChartContainer from './components/chart/ChartContainer'
import TopPeers from './components/peers/TopPeers.tsx'
import MarketFooter from './components/footer/MarketFooter.tsx'
import Price from './components/price/Price.tsx'
import Search from './components/search/Search'
import { connect } from 'react-redux'
import * as logo from './images/logo.png'

const App = props => (
  <div className='gradient'>
    <div className='flexbox-container'>
      <div>
        <img src={logo} alt='Adaptive Logo' className='logo' />
      </div>
    </div>

    <div className='flexbox-container searchbar-container'>
      <div className='searchbar'>
        <Search />
      </div>
      <div className='price-information'>
        <Price price={props.price} />
      </div>
    </div>

    <hr />

    <div className='Grid Grid--gutters Grid--1of3'>
      <div className='Grid-cell main'>
        <div>
          <ChartContainer chart={props.chart} loading={props.loading} />
        </div>
      </div>

      <div className='Grid-cell'>
        <div>
          <News/>
        </div>
      </div>
    </div>

    <div className='Grid Grid--gutters Grid--1of3'>
      <div className='Grid-cell main'>
        <KeyStats keystats={props.keystats} />
      </div>

      <div className='Grid-cell'>
        <div>
          <Overview/>
          <TopPeers peers={props.peers} />
        </div>
      </div>
    </div>
    <div>
      <footer className='footer'>
        <MarketFooter market={props.market} />
      </footer>
    </div>
  </div>
)

export default connect(
  state => ({
    input: state.input,
    error: state.error,
    keystats: state.keystats,
    chart: state.chart,
    peers: state.peers,
    loading: state.loading,
    market: state.market,
    price: state.price
  }),
  null
)(App)
