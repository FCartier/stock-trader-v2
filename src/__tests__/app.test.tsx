import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import App from '../App'

configure({ adapter: new Adapter() });

describe("app component", ()=> {
  test("app component should render properly", () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })
})