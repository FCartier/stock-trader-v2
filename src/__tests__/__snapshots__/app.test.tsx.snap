// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component app component should render properly 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColorBottom": "#16233B",
      "backgroundColorTop": "#1E355E",
      "headerColor": "#72a2e9",
      "marketFooterBottom": "#002050",
      "marketFooterTop": "#01255c",
      "textColor": "white",
    }
  }
>
  <div>
    <Styled(div)>
      <div
        className="flexbox-container"
      >
        <div>
          <Styled(img)
            alt="Adaptive Logo"
            src="test-file-stub"
          />
          <Styled(button)
            onClick={[Function]}
          >
            Change Theme
          </Styled(button)>
        </div>
      </div>
      <div
        className="flexbox-container searchbar-container"
      >
        <div
          className="searchbar"
        >
          <Connect(Search) />
        </div>
        <div
          className="price-information"
        >
          <Connect(Price) />
        </div>
      </div>
      <hr />
      <div
        className="flexbox-container"
      >
        <Connect(StockInfo) />
      </div>
      <div
        className="Grid Grid--gutters Grid--1of3"
      >
        <div
          className="Grid-cell main"
        >
          <div>
            <Connect(Component) />
          </div>
        </div>
        <div
          className="Grid-cell"
        >
          <div>
            <Connect(News) />
          </div>
        </div>
      </div>
      <div
        className="Grid Grid--gutters Grid--1of3"
      >
        <div
          className="Grid-cell main"
        >
          <Connect(KeyStats) />
        </div>
        <div
          className="Grid-cell"
        >
          <div>
            <Connect(Overview) />
            <Connect(TopPeers) />
          </div>
        </div>
      </div>
    </Styled(div)>
    <footer
      className="footer"
    >
      <Connect(MarketFooter) />
    </footer>
  </div>
</ThemeProvider>
`;
